File: src/commands/hello.js
import { ActionRowBuilder, ButtonBuilder, ButtonStyle } from 'discord.js';
import logger from '../utils/logger.js';

export const name = 'hello';
export const prefix = '!hello';

export async function execute(message) {
  if (message.guild) {
    logger.info(`Message from guild: ${message.guild.name}`, { guildId: message.guild.id });
  }

  const button = new ButtonBuilder()
    .setCustomId('open_modal')
    .setLabel('Open Modal')
    .setStyle(ButtonStyle.Primary);

  const row = new ActionRowBuilder().addComponents(button);

  await message.reply({ content: 'Click to open the modal', components: [row] });
}

export default { name, prefix, execute };

File: src/commands/hello.py
"""Hello command handler."""
import discord
from src.utils.logger import logger


NAME = 'hello'
PREFIX = '!hello'


async def execute(message: discord.Message):
    """Execute hello command."""
    if message.guild:
        logger.info(
            f"Message from guild: {message.guild.name}",
            {'guildId': message.guild.id}
        )

    button = discord.ui.Button(
        label='Open Modal',
        style=discord.ButtonStyle.primary,
        custom_id='open_modal'
    )

    view = discord.ui.View()
    view.add_item(button)

    await message.reply('Click to open the modal', view=view)
